h/static/scripts/widget-controller.js

```diff
@@ -343,6 +369,11 @@ module.exports = function WidgetController(
     return thread().totalChildren;
   };
 
+  $scope.filterAnnotationsByType = function (type) {
+    $scope.clearSelection();
```

Comment:
Clearing the selection when the current tab changes can be done in the reducer function that handles the state update when the selected tab changes. This will enforce that the selection is cleared when the tab is changed via another means.